src/provider/provider_os_memory_linux.c

```diff
@@ -32,6 +36,74 @@ umf_result_t os_translate_mem_protection_one_flag(unsigned in_protection,
     return UMF_RESULT_ERROR_INVALID_ARGUMENT;
 }
 
+umf_result_t os_translate_memory_flag(unsigned in_flag, unsigned *out_flag) {
+    switch (in_flag) {
+    case UMF_MEM_MAP_PRIVATE:
+        *out_flag = MAP_PRIVATE;
+        return UMF_RESULT_SUCCESS;
+    case UMF_MEM_MAP_SHARED:
+#ifdef __APPLE__
+        return UMF_RESULT_ERROR_NOT_SUPPORTED; // not supported on MacOSX
+#else
+        *out_flag = MAP_SHARED;
+        return UMF_RESULT_SUCCESS;
+#endif
+    }
+    return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+}
+
+int os_create_memory_fd(unsigned translated_memory_flag) {
+#ifdef __APPLE__
+    (void)translated_memory_flag; // unused
+    return 0;                     // ignored on MacOSX
+#else
+    // fd is created only for MAP_SHARED
+    if (translated_memory_flag != MAP_SHARED) {
+        return 0;
+    }
+
+    int fd = -1;
+
+#ifdef __NR_memfd_secret
+    // SYS_memfd_secret is supported since Linux 5.14
+    fd = syscall(SYS_memfd_secret, 0);
+    if (fd == -1) {
+        LOG_PDEBUG("memfd_secret()");
+    }
+    if (fd > 0) {
+        LOG_DEBUG("file descriptor created using memfd_secret()");
+        return fd;
+    }
+    // syscall(SYS_memfd_secret) can fail with errno == ENOTSYS (function not implemented).
+    // We should try to call syscall(SYS_memfd_create) in this case.
+#endif /* __NR_memfd_secret */
+
+#ifdef __NR_memfd_create
+    // SYS_memfd_create is supported since Linux 3.17, glibc 2.27
+    fd = syscall(SYS_memfd_create, "fd_name", 0);
+    if (fd == -1) {
+        LOG_PDEBUG("memfd_create()");
+    }
+    if (fd > 0) {
+        LOG_DEBUG("file descriptor created using memfd_create()");
+    }
+#endif /* __NR_memfd_create */
+
```

Comment:
could you add an fallback to error if neither SYS_memfd_create nor SYS_memfd_secret is supported?